base/utils.bal

```diff
@@ -203,3 +206,46 @@ public isolated function decodeFhirSearchParameters(string urlEncodedParams)
     }
     return arr;
 }
+public isolated function compareResources(ResourceCompareRequest resourceCompareRequest) returns error|http:Response {
+    http:Response response = new;
+    int score = 0;
+    foreach string fhirPathRule in resourceCompareRequest.fhirPaths {
+        fhirpath:FhirPathResult resultMapSourceResource = fhirpath:getFhirPathResult(<map<json>>resourceCompareRequest.sourceResource.toJson(), fhirPathRule);
+        fhirpath:FhirPathResult resultMapTargetResource = fhirpath:getFhirPathResult(<map<json>>resourceCompareRequest.targetResource.toJson(), fhirPathRule);
+        if !(resultMapSourceResource?.result is () || resultMapTargetResource?.result is ()) {
+            return error("No result found for the given FHIRPath expression in one of the resources");
+        }
+        if resultMapSourceResource.get(RESULT) is string && resultMapTargetResource.get(RESULT) is string {
+            string strResultSourceResource = <string>resultMapSourceResource.get(RESULT);
+            string strResultTargetResource = <string>resultMapTargetResource.get(RESULT);
+            if strResultSourceResource.equalsIgnoreCaseAscii(strResultTargetResource) {
+                score = score + 1;
+            } else {
+                response.setJsonPayload(false);
+                break;
+            }
+        } else {
+            if resultMapSourceResource.get(RESULT) == resultMapTargetResource.get(RESULT) {
+                score = score + 1;
+            } else {
+                response.setJsonPayload(false);
+                break;
+            }
+        }
+
+    }
+    if score == resourceCompareRequest.fhirPaths.length() {
+        response.setJsonPayload(true);
+    }
+    return response;
+}
+
+# Record to hold the resource details to be compared.
+public type ResourceCompareRequest record {|
```

Comment:
Shall we rename to ResourceCompareRequestData?

Reply:
Fixed in a6f84efbf7450ba0f7550ca40fd9f12a2f31121e